File: competitive_programming/codeforces/chat_room.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string s;
    cin >> s;
    string t = "";

    for (auto c : s) {
        if (c == 'h' && t.size() == 0) t += 'h';
        if (c == 'e' && t.size() == 1) t += 'e';
        if (c == 'l' && count(t.begin(), t.end(), 'l') == 1) t += 'l';
        if (c == 'l' && t.size() == 2) t += 'l';
        if (c == 'o' && t.size() == 4) t += 'o';
    }

    if (t == "hello") cout << "YES" << endl;
    else cout << "NO" << endl;
}

File: competitive_programming/codeforces/chat_room.py
s = input()

t = []
for c in s:
    if c == 'h' and len(t) == 0:
        t.append('h')
    if c == 'e' and len(t) == 1:
        t.append('e')
    if c == 'l' and t.count('l') == 1:
        t.append('l')
    if c == 'l' and len(t) == 2:
        t.append('l')
    if c == 'o' and len(t) == 4:
        t.append('o')
        break

t = ''.join(t)
if t == 'hello':
    print('YES')
else:
    print('NO')